File: python/test/ranking_api_test.py
import os
import sys
import unittest

sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '../')))

from maplestory_openapi.api.kms import MapleStoryApi

api_key = os.environ['API_KEY']  # Your API Key
api = MapleStoryApi(api_key)


# 랭킹 정보 조회
class RankingApi(unittest.IsolatedAsyncioTestCase):
    async def test_overall_ranking(self):
        response = await api.get_overall_ranking()
        print(response)

    async def test_union_ranking(self):
        response = await api.get_union_ranking()
        print(response)

    async def test_guild_ranking(self):
        response = await api.get_guild_ranking()
        print(response)

    async def test_dojang_ranking(self):
        response = await api.get_dojang_ranking()
        print(response)

    async def test_seed_ranking(self):
        response = await api.get_theseed_ranking()
        print(response)

    async def test_achievement_ranking(self):
        response = await api.get_achievement_ranking()
        print(response)


if __name__ == '__main__':
    unittest.main()

File: js/test/rankingApi.test.ts
import process from 'process';

import { toString } from './utils';
import { MapleStoryApi } from '../src/maplestory/api/mapleStoryApi';

const apiKey = process.env.API_KEY!; // Your API Key
const api = new MapleStoryApi(apiKey);

describe('랭킹 정보 조회', () => {
  test('종합 랭킹 정보 조회', async () => {
    const response = await api.getOverallRanking();

    console.log(toString(response));
  });

  test('유니온 랭킹 정보 조회', async () => {
    const response = await api.getUnionRanking();

    console.log(toString(response));
  });

  test('길드 랭킹 정보 조회', async () => {
    const response = await api.getGuildRanking();

    console.log(toString(response));
  });

  test('무릉도장 랭킹 정보 조회', async () => {
    const response = await api.getDojangRanking();

    console.log(toString(response));
  });

  test('더 시드 랭킹 정보 조회', async () => {
    const response = await api.getSeedRanking();

    console.log(toString(response));
  });

  test('업적 랭킹 정보 조회', async () => {
    const response = await api.getAchievementRanking();

    console.log(toString(response));
  });
});
